internal/components/componentlist.go

```diff
@@ -0,0 +1,132 @@
+package components
```

Comment:
Just a small suggestion: we could name the package `component` and then remove the `Component` prefix from all of the types. It would read nicer: `component.List`, `component.Definition`, `component.ListData`, etc.